consumer/retry_counter_test.go

```diff
@@ -0,0 +1,44 @@
+package consumer
+
+import (
+	"github.com/stretchr/testify/assert"
+	"testing"
+)
+
+func TestRetryCounter(main *testing.T) {
+	main.Run("RetryCount is initially zero", func(t *testing.T) {
```

Comment:
All test cases has the prefix `RetryCount` which is inferred in the test runs already. This test will be named `TestRetryCounter/RetryCount_is_initially_zero`.
Maybe drop the `RetryCount` prefix so it becomes `TestRetryCounter/is_initially_zero`

```suggestion
	main.Run("is initially zero", func(t *testing.T) {
```

Same with the rest of the tests.